### DISCUSSION TITLE
Doctrine/dbal - php fatal error

### DISCUSSION THREAD

**Discussion by Saliva-sys** [ddev/ddev#6576] [Q&A] [answered]
hello, I want to ask you for help. I have an eshop through prestashop, but everything stopped working. It gives me fatal error 500. I can't open the page or log in to the admin interface. In the error logs, the error "PHP Fatal error: Declaration of Doctrine\\DBAL\\Driver\\PDOConnection::query() must be compatible with PDO::query(string $query, ?int $fetchMode = null, mixed . ..$fetchModeArgs): PDOStatement|false in /data/b/a/baaeb66d-3318-4d95-b15c-650617fafdb5/amcmcars.sk/sub/eshop/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection .php on line 70". Can you please advise if it is related to the version of php and Doctrine/dbal? When I looked, the Doctrine/dbal version I have in prestashop is v2.9.3. Need to update Doctrine? Thank you in advance for your help

**Comment by rfay** [comment, marked as answer]
I'm afraid this is not a DDEV question at all. 

As far as I can tell, the current version of PrestaShop is 8.2.0, see https://github.com/PrestaShop/PrestaShop

So you must be many years behind?

In this I see that you need to downgrade to PHP 7.4 maybe? https://www.prestashop.com/forums/topic/1069631-solved-update-ps-17-with-php-8-fatall-error/#google_vignette

This seems to say the same: https://github.com/PrestaShop/PrestaShop/issues/30642

So `ddev config --php-version=7.4 && ddev restart` and see what happens

**Reply by Saliva-sys** [reply]
I apologize, I'm only here for a short time, I'm just getting my bearings on how it works here. I'll be careful. Anyway, thanks for the reply, it solved the problem. Many thanks
